### PR TITLE
Respect disabled config generation in lombok plugin

### LINKED ISSUE #46: Lombok plugin "generateLombokConfig.enabled = false" is ignored
I have the following build file:
```kotlin
...
subprojects {
    apply(rootProject.file("gradle/lombok.gradle"))
    apply(rootProject.file("gradle/checker-framework.gradle"))
...
}
...
```
lombok.gradle:
```groovy
apply plugin: 'io.freefair.lombok'
generateLombokConfig.enabled = false

lombok {
    version = '1.18.8'
}
```
checker-framework.gradle:
```groovy
apply plugin: 'org.checkerframework'

checkerFramework {
    excludeTests = true
    checkers = [
            'org.checkerframework.checker.nullness.NullnessChecker',
            'org.checkerframework.checker.units.UnitsChecker'
    ]
}
```
For some reason, when I enable `checker-framework.gradle` the lombok's `generateLombokConfig.enabled = false` is ignored and I have bunch of lombok.config generated files all over my modules.

### PR DESCRIPTION
**Pull request by kelloggm**
Should fix #46 

The `io.freefair.lombok` plugin includes an option to disable config file generation, which we were not respecting (because some checkers - like the object construction checker - rely on certain options being set, we were generating an appropriate config automatically).

This behavior is now guarded in the same manner used by the [lombok plugin itself](https://github.com/freefair/gradle-plugins/blob/3.8.0/lombok-plugin/src/main/java/io/freefair/gradle/plugins/lombok/LombokPlugin.java#L67).

### CODE REVIEW

**Comment by kelloggm** [comment]
@sanyarnd, can you confirm that this change is what you were looking for when you reported #46? (I haven't yet released `0.3.31`, but if you build the plugin locally, you can test it after running `./gradlew publishToMavenLocal` in the plugin repo)

File: src/main/groovy/org/checkerframework/gradle/plugin/CheckerFrameworkPlugin.groovy [hunk 1]
```diff
@@ -62,7 +62,11 @@ final class CheckerFrameworkPlugin implements Plugin<Project> {
 
         // Ensure that the lombok config is set to emit @Generated annotations
         lombokPlugin.configureForJacoco()
-        lombokPlugin.generateLombokConfig.get().generateLombokConfig()
+        // If config generation is enabled, automatically produce an appropriate config file.
+        def generateLombokConfig = lombokPlugin.generateLombokConfig.get()
+        if (generateLombokConfig.isEnabled()) {
```
Comment (mernst) [inline src/main/groovy/org/checkerframework/gradle/plugin/CheckerFrameworkPlugin.groovy]: What are the consequences for the plugin if this test evaluates to false?
That is, I presume that the Lombok configuration does something useful.  Is that functionality lost, or is it achieved in some other way?

**Comment by sanyarnd** [comment]
@kelloggm the patch has no effect, I've attached the MWE to ease your testing.
You can see that once you import the project (and every time you re-import it due to build file changes) `lombok.config` is generated inside `submodule` directory.

[sample.zip](https://github.com/kelloggm/checkerframework-gradle-plugin/files/3596761/sample.zip)

File: src/main/groovy/org/checkerframework/gradle/plugin/CheckerFrameworkPlugin.groovy (hunk 1 above)
Comment (kelloggm) [reply src/main/groovy/org/checkerframework/gradle/plugin/CheckerFrameworkPlugin.groovy]: The object construction checker, if run, will not issue warnings on incorrect Lombok builders. As far as I know, that's the only checker that relies on this behavior.

**Comment by kelloggm** [comment]
@sanyarnd, I've tested it with your sample project - thanks for the MWE! The problem appears to be that in the MWE, the lombok plugin is applied *after* the Checker Framework plugin, so when the CF plugin decides to generate a lombok config, the options on the lombok plugin aren't yet available. So, in the short term, switching the order of the plugin applications in the `build.gradle.kts` files should remove the problem (along with using `0.3.31`):

```
-    apply(from = "$rootDir/gradle/checker-framework.gradle")
-    apply(from = "$rootDir/gradle/lombok.gradle")
+    apply(from = "$rootDir/gradle/lombok.gradle")
+    apply(from = "$rootDir/gradle/checker-framework.gradle")
```

My next step will be to see if I can defer making that decision until all plugins are configured.

**Pull request merged**